### ISSUE TITLE
Request for summary lables

### ISSUE THREAD

**Issue by xli4217** [tensorforce/tensorforce#275]
Is there a place where the labels for the summary_spec are summarized? Or is it necessary to go into the code to find them. Also, the names for the scalars are a bit confusing or am I using this feature wrong. Thanks.

**Comment by ImpulseAdventure** [comment]
@Mazecreator created a nice overview in #265 very recently. Have a look at [docs/summary_spec.md](https://github.com/reinforceio/tensorforce/blob/master/docs/summary_spec.md).

**Comment by xli4217** [comment]
I see, this is very helpful. But if I want additional logging that's specific to an algorithm (like policy entropy or KL), is there currently a convenient way to do so?

**Comment by Mazecreator** [comment]
@xli4217 try the "regularization" to see if that gives you what you want.  This should include some of the entropy information.  The details you are looking for may not be available separately when the regularization information is consolidated.

One thing that was not included in that spec, is the ability to add the summary_labels spec directly into a network_spec dict() entry.  This will only generate those values for that particular layer.  I think the only two that work in the dict() are summary_labels=['relu','activations'], others can be included but are unlikely to have an effect on TensorBoard displays.

**Comment by xli4217** [comment]
It would be nice to be able to log some of the optimizers' outputs, are there plans for users to create custom summary_labels?

**Comment by AlexKuhnle** [comment]
If you are interested in specific values to be logged, we (or you in a PR) can add them. What values would you want to be able to log?

**Comment by xli4217** [comment]
For example the entropy of the current Gaussian policy, the explained variance, or KL-divergence between the policy before and after an update. I see there are such functions in the Gaussian class like tf_entropy(distr_params) and tf_kl_divergence(distr_params1, distr_params2), but those functions are quite a distance away from the use interface, I was wondering if there's a good way to get these information in tensorboard as distributions, if not then to expose them to a runner class and log to csv files.

**Comment by AlexKuhnle** [comment]
We can look into that. Entropy is no problem, KL divergence requires additional computations, but should also be possible. Tensorboard is probably the best way to record all this.

Could you rename the issue to something more descriptive, like 'request for summary labels' or so? And then feel free to add whatever else you think is generally useful.

**Comment by xli4217** [comment]
Thanks, in addition to entropy and KL, it would also be nice to be able to log at each step the state_value predicted by the baseline, td_residual, the learning rate if a schedule is used, etc. These intermediate values provide valuable insights to the learning process and often help with hyperparameter tuning.

In general it would be helpful to provide an interface consistent with what tensorforce is using internally for logging that users can use to log entities for algorithms that they implement.

**Comment by AlexKuhnle** [comment]
Summaries work with the new branch (`major-revision`), which will be merged with master soon, hence already closing summary-related issues. Currently supported labels:
```python
"bernoulli", "beta", "categorical", "distributions", "dropout", "entropy", "gaussian", "graph",
"loss", "losses", "objective-loss", "regularization-loss", "relu", "updates", "variables"
```
Others will be added, and please feel free to post specific requests here.
